### PR TITLE
UGENE-7623 CRASH if Sanger reads contain gaps or "N"s only

### PR DESCRIPTION
**Pull request by DmitriiSukhomlinov**

### CODE REVIEW

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 1]
```diff
@@ -2769,6 +2769,47 @@ GUI_TEST_CLASS_DEFINITION(test_7616) {
...
+
+    // Open "Trim and аlign Sanger reads" sample
+    class Scenario : public CustomScenario {
+    public:
+        void run(HI::GUITestOpStatus& os) {
+            // Expected state: "Trim and Align Sanger Reads" dialog has appered
+            QWidget* wizard = GTWidget::getActiveModalWidget(os);
+            const QString expectedTitle = "Map Sanger Reads to Reference";
+            const QString actualTitle = wizard->windowTitle();
+            CHECK_SET_ERR(expectedTitle == actualTitle, QString("Wizard title is incorrect: expected '%1', got '%2'").arg(expectedTitle).arg(actualTitle));
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Remove this check: ```WizardFiller(os, "Map Sanger Reads to Reference"``` guarantees that the scenario is triggered for the correct dialog.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 2]
```diff
@@ -2769,6 +2769,47 @@ GUI_TEST_CLASS_DEFINITION(test_7616) {
...
+            GTWidget::click(os, wizard);
+
+            // Select Reference .../test/general/_common_data/sanger/reference.gb
+            GTUtilsWizard::setParameter(os, "Reference", testDir + "_common_data/sanger/reference.gb");
+
+            // Push Next
+            GTUtilsWizard::clickButton(os, GTUtilsWizard::Next);
+
+            // On page "Input Sanger reads" add: .../test/general/_common_data/sanger/sanger_01.fa and click "Next" button
+            GTUtilsWizard::setInputFiles(os, { { testDir + QString("_common_data/sanger/sanger_01.fa") } });
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: This is a new file added for this test: use a better name for it like n_and_gaps.fa

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 3]
```diff
@@ -2769,6 +2769,47 @@ GUI_TEST_CLASS_DEFINITION(test_7616) {
...
+GUI_TEST_CLASS_DEFINITION(test_7623) {
+    GTLogTracer logTracer;
+
+    // Select "Tools>Workflow Designer"
+    GTUtilsWorkflowDesigner::openWorkflowDesigner(os);
+
+    // Open "Trim and аlign Sanger reads" sample
+    class Scenario : public CustomScenario {
+    public:
+        void run(HI::GUITestOpStatus& os) {
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Add override

File: src/plugins/external_tool_support/src/blast/align_worker_subtasks/BlastAlignToReferenceTask.cpp [hunk 4]
```diff
@@ -142,6 +142,7 @@ void BlastAlignToReferenceTask::prepare() {
     settings.gapOpenCost = 2;
     settings.gapExtendCost = 2;
     blastQuerySequenceIndexByReadIndex.resize(reads.size());
+    bool allReadsAreEmpty = true;
```
Comment (yalgaer) [inline src/plugins/external_tool_support/src/blast/align_worker_subtasks/BlastAlignToReferenceTask.cpp]: You do not need this variable at all. You can check that `settings.querySequences` are empty or `settings.alphabet` is nullptr

**Review by yalgaer — CHANGES_REQUESTED** [review]

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp (hunk 1 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: OK

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp (hunk 2 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: OK

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp (hunk 3 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: OK

File: src/plugins/external_tool_support/src/blast/align_worker_subtasks/BlastAlignToReferenceTask.cpp (hunk 4 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/external_tool_support/src/blast/align_worker_subtasks/BlastAlignToReferenceTask.cpp]: OK

**Review by smlnkcv — APPROVED** [review]

**Review by yalgaer — APPROVED** [review]

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 5]
```diff
@@ -2769,6 +2769,39 @@ GUI_TEST_CLASS_DEFINITION(test_7616) {
...
+    public:
+        void run(HI::GUITestOpStatus& os) override {
+            // Select Reference .../test/general/_common_data/sanger/reference.gb
+            GTUtilsWizard::setParameter(os, "Reference", testDir + "_common_data/sanger/reference.gb");
+
+            // Push Next
+            GTUtilsWizard::clickButton(os, GTUtilsWizard::Next);
+
+            // On page "Input Sanger reads" add: .../test/general/_common_data/sanger/n_and_gaps.fa and click "Next" button
+            GTUtilsWizard::setInputFiles(os, { { testDir + QString("_common_data/sanger/n_and_gaps.fa") } });
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp:2789]: nit: the file contains Ns only, no gaps

**Pull request merged**